server/coordinator/procedure/factory.go

```diff
@@ -60,6 +60,10 @@ type DropTableRequest struct {
 	OnFailed    func(error) error
 }
 
+func (d DropTableRequest) IsPartitionTable() bool {
+	return d.SourceReq.PartitionTableInfo != nil
```

Comment:
Check `partitionTableInfo.subTableNames`, you can refer to `CreateTableRequest`.